lib/core/src/Cardano/Wallet/Primitive/Model.hs

```diff
@@ -386,26 +401,32 @@ blockchainParameters (Wallet _ _ _ bp) = bp
 -- linked to the wallet by their inputs.
 --
 -- In order to identify transactions that are ours, we do therefore look for
--- known inputs and known outputs. However, we can't naievely look at the domain
+-- known inputs and known outputs. However, we can't naively look at the domain
 -- of the utxo constructed from all outputs that are ours (as the specification
 -- would suggest) because some transactions may use outputs of a previous
--- transaction within the same block as an input. Therefore, Looking solely at
+-- transaction within the same block as an input. Therefore, looking solely at
 -- the final 'dom (UTxO ⊳ oursOuts)', we would be missing all intermediate txs
 -- that happen from _within_ the block itself.
 --
--- As a consequence, we do have to traverse the block, and look at transaction
+-- As a consequence, we do have to traverse the block, and look at transactions
 -- in order, starting from the known inputs that can be spent (from the previous
--- UTxO) and, collect resolved tx outputs that are ours as we apply transactions.
+-- UTxO) and collect resolved tx outputs that are ours as we apply transactions.
 prefilterBlock
-    :: (IsOurs s Address)
+    :: (IsOurs s Address, IsOurs s ChimericAccount)
     => Block
     -> UTxO
     -> s
-    -> (([(Tx, TxMeta)], UTxO), s)
+    -> ((FilteredBlock, UTxO), s)
 prefilterBlock b u0 = runState $ do
-    (ourTxs, ourU) <- foldM applyTx (mempty, u0) (transactions b)
-    return (ourTxs, ourU)
+    delegations <- fmap snd <$> filterM isOurDelegation (b ^. #delegations)
```

Comment:
**[suggestion]**

Use [mapMaybeM](https://hackage.haskell.org/package/extra-1.6.18/docs/Control-Monad-Extra.html#v:mapMaybeM)

**Why**

Avoid traversing the list twice.

Reply:
@KtorZ wrote:
> **[suggestion]**
> Use [mapMaybeM](https://hackage.haskell.org/package/extra-1.6.18/docs/Control-Monad-Extra.html#v:mapMaybeM)

I've followed this suggestion. See:

 https://github.com/input-output-hk/cardano-wallet/compare/6ffbaebf325b9fd35caf69a86424894686a10992..96c00ac44b5973ebbf9b39129a5114c9f611b848